soc/soc_legacy/arm/atmel_sam/sam3x/Kconfig.soc

```diff
@@ -1,89 +0,0 @@
-# Atmel SAM3X MCU series
-
-# Copyright (c) 2017 Justin Watson
-# Copyright (c) 2016 Intel Corporation.
-# Copyright (c) 2014-2015 Wind River Systems, Inc.
-# SPDX-License-Identifier: Apache-2.0
-
-choice
-	prompt "Atmel SAM3X MCU Selection"
-	depends on SOC_SERIES_SAM3X
-
-	config SOC_PART_NUMBER_SAM3X4C
-		bool "SAM3X4C"
-	config SOC_PART_NUMBER_SAM3X4E
-		bool "SAM3X4E"
-	config SOC_PART_NUMBER_SAM3X8C
-		bool "SAM3X8C"
-	config SOC_PART_NUMBER_SAM3X8E
```

Comment:
the SoC part numbers seems to have gone missing.

It's used here:
https://github.com/zephyrproject-rtos/zephyr/blob/00deb8dfb10fa7da8dfcb3719fb93fe410924b68/samples/drivers/lcd_hd44780/src/main.c#L74

Please ensure the SoC part numbers are still available, as well as checking other settings from this files, as I believe they should be preserved.

Reply:
I renamed as the below snip:

```diff
-config SOC_PART_NUMBER_SAME70Q21B
+config SOC_SAME70Q21B
	bool
	select SOC_SERIES_SAME70
	select SOC_ATMEL_SAME70_REVB

config SOC
	default "same70j19" if SOC_SAME70J19
```